yb-voyager/cmd/analyzeSchema.go

```diff
@@ -404,17 +410,145 @@ func checkStmtsUsingParser(sqlInfoArr []sqlInfo, fpath string, objType string) {
 		}
 		if isAlterTable {
 			reportAlterTableVariants(alterTableNode, sqlStmtInfo, fpath, objType)
 			reportExclusionConstraintAlterTable(alterTableNode, sqlStmtInfo, fpath)
+			reportDeferrableConstraintAlterTable(alterTableNode, sqlStmtInfo, fpath)
 		}
 		if isCreateIndex {
 			reportCreateIndexStorageParameter(createIndexNode, sqlStmtInfo, fpath)
 		}
 	}
 }
 
+func reportXMLAndXIDDatatype(createTableNode *pg_query.Node_CreateStmt, sqlStmtInfo sqlInfo, fpath string) {
+	schemaName := createTableNode.CreateStmt.Relation.Schemaname
+	tableName := createTableNode.CreateStmt.Relation.Relname
+	columns := createTableNode.CreateStmt.TableElts
+	fullyQualifiedName := lo.Ternary(schemaName != "", schemaName+"."+tableName, tableName)
+	for _, column := range columns {
+		/*
+			e.g. CREATE TABLE test_xml_type(id int, data xml);
+			relation:{relname:"test_xml_type" inh:true relpersistence:"p" location:15} table_elts:{column_def:{colname:"id"
+			type_name:{names:{string:{sval:"pg_catalog"}} names:{string:{sval:"int4"}} typemod:-1 location:32}
+			is_local:true location:29}} table_elts:{column_def:{colname:"data" type_name:{names:{string:{sval:"xml"}}
+			typemod:-1 location:42} is_local:true location:37}} oncommit:ONCOMMIT_NOOP}}
+
+			here checking the type of each column as type definition can be a list names for types which are native e.g. int
+			it has type names - [pg_catalog, int4] both to determine but for complex types like text,json or xml etc. if doesn't have
+			info about pg_catalog. so checking the 0th only in case XML/XID to determine the type and report
+		*/
+		if column.GetColumnDef() != nil {
+			typeName := ""
+			if len(column.GetColumnDef().GetTypeName().GetNames()) > 0 {
+				typeName = column.GetColumnDef().GetTypeName().GetNames()[0].GetString_().Sval // 0th index as for these non-native types pg_catalog won't be present on first location
+			} 
+			colName := column.GetColumnDef().GetColname()
+			reason := fmt.Sprintf("%s - %s on column - %s", UNSUPPORTED_DATATYPE, typeName, colName)
+			if typeName == "xml" {
+				summaryMap["TABLE"].invalidCount[sqlStmtInfo.objName] = true
+				reportCase(fpath, reason, "https://github.com/yugabyte/yugabyte-db/issues/1043",
+					"Data ingestion is not supported for this type in YugabyteDB so handle this type in different way. Refer link for more details - <LINK DOC>",
+					"TABLE", fullyQualifiedName, sqlStmtInfo.formattedStmt, true)
+			}
+			if typeName == "xid" {
+				summaryMap["TABLE"].invalidCount[sqlStmtInfo.objName] = true
+				reportCase(fpath, reason, "https://github.com/yugabyte/yugabyte-db/issues/15638",
+					"Functions for this type e.g. txid_current are not supported in YugabyteDB yet",
+					"TABLE", fullyQualifiedName, sqlStmtInfo.formattedStmt, true)
+			}
+		}
+	}
+}
+
+var deferrableConstraintsList = []pg_query.ConstrType{
+	pg_query.ConstrType_CONSTR_ATTR_DEFERRABLE,
+	pg_query.ConstrType_CONSTR_ATTR_DEFERRED,
+	pg_query.ConstrType_CONSTR_ATTR_IMMEDIATE,
```

Comment:
How is "IMMEDIATE" a deferrable constraint?

Reply:
this is for `DEFERRABLE INITIALLY IMMEDIATE` which means that normally the constraint is checked immediately after each statement but user can defer the constraint within a transaction if necessary.

Reply:
pls add a comment there